src/Orckestra.Composer.Cart/Providers/Order/EditingOrderProvider.cs

```diff
@@ -53,6 +57,97 @@ public virtual async Task<bool> CanEdit(Overture.ServiceModel.Orders.Order order
             return isOrderEditable;
         }
 
+        public virtual async Task<OrderCancellationStatusViewModel> GetCancellationStatus(Overture.ServiceModel.Orders.Order order)
+        {
+            if (!ValidateOrderForCancel(order))
+            {
+                return new OrderCancellationStatusViewModel(){CanCancel = false, CancellationPending = false};
+            }
+            var orderFulfillmentState = await OrderRepository.GetOrderFulfillmentStateAsync(new GetOrderFulfillmentStateParam
+            {
+                OrderId = order.Id,
+                ScopeId = order.ScopeId
+            }).ConfigureAwait(false);
+
+            if (OrderHistoryConfiguration.CompletedOrderStatuses.Contains(orderFulfillmentState.Status))
+            {
+                return new OrderCancellationStatusViewModel() { CanCancel = false, CancellationPending = false };
+            }
+
+            switch (orderFulfillmentState.IsCancelable)
+            {
+                case true when !orderFulfillmentState.IsProcessing && orderFulfillmentState.ShipmentFulfillmentStates.All(item => item.AllowedStatusChanges?.Contains(Constants.OrderStatus.Canceled) ?? false):
+                    return new OrderCancellationStatusViewModel() { CanCancel = true, CancellationPending = false };
+                case false when orderFulfillmentState.IsProcessing && HasOrderShipmentFulfillmentStatesCancelMessage(orderFulfillmentState, order.Id):
+                    return new OrderCancellationStatusViewModel() { CanCancel = false, CancellationPending = true };
+                default:
+                    return new OrderCancellationStatusViewModel();
+            }
+        }
+
+        private bool HasOrderShipmentFulfillmentStatesCancelMessage(OrderFulfillmentState orderFulfillmentState, string orderId)
+        {
+            if (orderFulfillmentState?.ShipmentFulfillmentStates == null) return false;
+
+            return orderFulfillmentState.ShipmentFulfillmentStates.Any(item =>
+                item.Messages?.Exists(el => el.MessageId == orderId
+                                            && el.PropertyBag[Constants.DefaultOrderCancellationMessage] is DateTime
+                                            && (DateTime)el.PropertyBag[Constants.DefaultOrderCancellationMessage] >
+                                            DateTime.UtcNow.AddMinutes(-10)) ?? false);
+
+        }
+
+        private bool ValidateOrderForCancel(Overture.ServiceModel.Orders.Order order)
+        {
+            if (order?.Cart?.Shipments == null
+                || OrderHistoryConfiguration.CompletedOrderStatuses.Contains(order.OrderStatus)
+                || string.IsNullOrWhiteSpace(order.OrderStatus))
```

Comment:
you need check string.IsNullOrWhiteSpace(order.OrderStatus)) before the .Contains(order.OrderStatus)